src/MizarData/BaselineAndComparisonTest.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/container/flat_hash_map.h>
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "BaselineAndComparisonHelper.h"
+#include "MizarData/BaselineAndComparison.h"
+
+namespace orbit_mizar_data {
+
+constexpr size_t kFunctionNum = 3;
+constexpr std::array<uint64_t, kFunctionNum> kBaselineFunctionAddresses = {0xF00D, 0xBEAF, 0xDEAF};
+constexpr std::array<uint64_t, kFunctionNum> kComparisonFunctionAddresses = {0x0FF, 0xCAFE, 0xDEA};
+static const std::vector<std::string> kBaselineFunctionNames = {"foo()", "bar()", "biz()"};
+static const std::vector<std::string> kComparisonFunctionNames = {"foo()", "bar()", "fiz()"};
+
+template <typename Container>
+auto Commons(const Container& a, const Container& b) {
+  typedef typename Container::value_type E;
+  absl::flat_hash_set<E> a_set(std::begin(a), std::end(a));
```

Comment:
By using `std::begin` you implicitly assume that `a` is a container living in the `std` namespace or has `begin` and `end` member functions..

For writing this properly generic you would do this (Search for using-ADL-dance internally to get some more details):
```cpp
using std::begin;
using std::end;
absl::flat_hash_set<E> a_set(begin(a), end(a));
```

This uses ADL to find the begin/and for containers that don't have `begin/end` member functions and fall back to `std::begin` if non exist.